frontend/javascripts/oxalis/model/reducers/flycam_reducer.js

```diff
@@ -105,13 +105,24 @@ function resetMatrix(matrix: Matrix4x4, dataSetScale: Vector3) {
   return newMatrix;
 }
 
-function moveReducer(state: OxalisState, vector: Vector3): OxalisState {
+function moveReducer(state: OxalisState, vector: Vector3, clampToEdge: boolean): OxalisState {
   const matrix = cloneMatrix(state.flycam.currentMatrix);
   if (!vector.includes(NaN)) {
     matrix[12] += vector[0];
     matrix[13] += vector[1];
     matrix[14] += vector[2];
   }
+  if (clampToEdge) {
+    if (vector[2] > 0) {
+      // If the direction is incrementing, we clamp to .0 so that subsequent movements
+      // go from .0 to 0.999 before the next slice is shown.
+      matrix[14] = Math.floor(matrix[14]);
+    } else {
+      // If the direction is decrementing, we clamp to .999 so that subsequent movements
+      // go from .999 to .0 before the next slice is shown.
+      matrix[14] = Math.floor(matrix[14]) + 0.999;
+    }
+  }
```

Comment:
> As far as I can see, this special behavior (first event changes coordinate, subsequent events increment in small steps) is only performed for "depth" related movement (moveZ has a suboptimal name as the z refers to the dimension which is orthogonal to the current viewport). Thus, shortcuts as f, d, space etc should be covered. Moving via arrow keys shouldn't be affected as these always use the normal move speed.

Yes agreed, that's what I meant as well :) However, wouldn't that mean that you would need to look up which component of `vector` is the "depth" one, here (depending on the `activePlaneId`)? Currently you're always accessing the true z component by using `vector[2]` and `matrix[14]`, but it should vary depending on which is the active viewport. See for example the `MOVE_FLYCAM_ORTHO` reducer which also does something like `const dim = Dimensions.getIndices(planeId)[2];` for the dynamic space direction.

My suggestion would be to add the active `planeId` to the function signature and then something like
```suggestion
  if (clampToEdge) {
    const dim = Dimensions.getIndices(planeId)[2];
    if (vector[dim] > 0) {
      // If the direction is incrementing, we clamp to .0 so that subsequent movements
      // go from .0 to 0.999 before the next slice is shown.
      matrix[12 + dim] = Math.floor(matrix[12 + dim]);
    } else {
      // If the direction is decrementing, we clamp to .999 so that subsequent movements
      // go from .999 to .0 before the next slice is shown.
      matrix[12 + dim] = Math.floor(matrix[12 + dim]) + 0.999;
    }
  }
```

Reply:
Ahh, you are obviously completely right. Thanks for not getting sidetracked by my stubborn assumption that my implementation was working for all dimensions :see_no_evil: Also thank you for coding the fix!